src/org/zaproxy/zap/extension/frontEndScanner/FrontEndScannerAPI.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.frontEndScanner;
+
+import java.util.Locale;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+public class FrontEndScannerAPI extends ApiImplementor {
+    // TODO shouldnt allow unsafe-inline styles - need to work out where they are being used
+    protected static final String CSP_POLICY =
+        "default-src 'none'; script-src 'self'; connect-src https://zap wss://zap; frame-src 'self'; img-src 'self' data:; "
+        + "font-src 'self' data:; style-src 'self' 'unsafe-inline' ;";
+
+    private static final String PREFIX = "frontEndScanner";
+
+    private ExtensionFrontEndScanner extension;
+
+    private static final Logger LOGGER = Logger.getLogger(FrontEndScannerAPI.class);
```

Comment:
Not used.

Reply:
Should I though? Do you have any guide about logging?
  * Where to put `LOGGER.debug()` statements?
  * Should I use the `LOGGER.error()` in `catch` clauses?

Reply:
Typically LOGGER.error should only be used if something happens which indicates a bug in the code, ie something we can fix. Ideally we should never get any ERRORs when running ZAP, although we still do ;)
Put debug statement in iff you think it will help you (or someone else) work out whats going wrong.